clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -1,33 +1,166 @@
+import static org.mockito.ArgumentMatchers.any;
+
+import org.mockito.Answers;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedConstruction.Context;
+
+import io.lakefs.clients.api.ApiException;
+import io.lakefs.clients.api.ObjectsApi;
+import io.lakefs.clients.api.RepositoriesApi;
+import io.lakefs.clients.api.StagingApi;
+import io.lakefs.clients.api.model.ObjectStats;
+import io.lakefs.clients.api.model.ObjectStatsList;
+import io.lakefs.clients.api.model.Repository;
+import io.lakefs.clients.api.model.StagingLocation;
+
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Random;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Protocol;
+import com.amazonaws.SDKGlobalConfiguration;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.S3ClientOptions;
+import com.amazonaws.services.s3.model.CreateBucketRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.aventrix.jnanoid.jnanoid.NanoIdUtils;
 
 public class LakeFSFileSystemTest {
+    protected static final Logger LOG = LoggerFactory.getLogger(LakeFSFileSystemTest.class);
+    protected final LakeFSFileSystem fs = new LakeFSFileSystem();
+
+    protected LakeFSClient lfsClient;
+    protected ObjectsApi objectsApi;
+    protected RepositoriesApi repositoriesApi;
+    protected StagingApi stagingApi;
+
+    protected AmazonS3 s3Client;
 
-    private LakeFSFileSystem fs;
+    protected String s3Base;
+    protected String s3Bucket;
+
+    private static final DockerImageName MINIO = DockerImageName.parse("minio/minio:RELEASE.2021-05-16T05-32-34Z");
+    protected static final String S3_ACCESS_KEY_ID = "AKIArootkey";
+    protected static final String S3_SECRET_ACCESS_KEY = "secret/minio/key=";
+
+    protected static final Regions REGION = Regions.US_EAST_1;
+
+    @Rule
+    public final GenericContainer s3 = new GenericContainer(MINIO.toString()).
+        withCommand("minio", "server", "/data").
+        withEnv("MINIO_ROOT_USER", S3_ACCESS_KEY_ID).
+        withEnv("MINIO_ROOT_PASSWORD", S3_SECRET_ACCESS_KEY).
+        //        withEnv("MINIO_REGION_NAME", REGION.getName()).
+        withEnv("MINIO_DOMAIN", "s3.local.lakefs.io").
+        withExposedPorts(9000);
+
+    protected static String makeS3BucketName() {
+        String slug = NanoIdUtils.randomNanoId(NanoIdUtils.DEFAULT_NUMBER_GENERATOR,
+                                               "abcdefghijklmnopqrstuvwxyz-0123456789".toCharArray(), 14);
```

Comment:
Suggestion: remove the '-' from the possible characters. based on the name bucket-x it looks like we use it as a separator.

Reply:
Not done.  The (only) reason to add "-x" is to avoid ending the bucket name with a dash, which is "of course" not allowed for some buckets sometimes.

Reply:
so why use '-' at the first place with nano?